paper.md

```diff
@@ -99,6 +99,45 @@ We also implement some features to improve robustness and flexibility of the ens
 
 * The TOML-file interface defined in the \texttt{UQParameters} module allows non-intrusive use of \texttt{EnsembleKalmanProcesses.jl} through \text{TOML} files, which are widely used for configuration files and easily read in any programming language. Given the computer model to calibrate and prior distributions on the parameters, \texttt{EnsembleKalmanProcesses.jl} reads these distributions from a file and, after an iteration of the ensemble Kalman algorithm, writes each member of the updated ensemble to a parameter file. Each of these parameter files can be then read individually to initiate the ensemble of the computer model for the next iteration.
 
+# Pedagogical example
+
+In this example, the complex computer code simulates a sinusoid curve $$f(A,v) = A\sin(\phi + t) + v, \ \forall t \in [0,2\pi],$$ with a random phase shift $\phi$. We assume that we cannot access $f$, but we can access some informative quantities through the map $$G(A,v) = [\max f(A,v) - \min f(A,v), \mathrm{mean} f(A,v)].$$
+
+We are given one sample measurement of $G$, polluted by Gaussian noise $N(0,\Gamma)$, and call this $y$. Our task is to deduce the most likely amplitude $A$ and vertical shift $v$ of the curve that produced the $y$. 
+
+
+We can encode prior information mathematically in the code:
+```julia
+# A is positive, has likely value 2 with standard deviation 1
+# v has likely value 0 with standard deviation 5
+prior_A = constrained_gaussian("amplitude", 2, 1, 0, Inf)
+prior_v = constrained_gaussian("vert_shift", 0, 5, -Inf, Inf)
+prior = combine_distributions([prior_A, prior_v])
+```
+
+To use a basic ensemble method we require one more parameter, the size of the ensemble, which we take to be `N_ensemble = 5`. We now begin solving the problem, by creating the initial ensemble from our prior, and selecting the `Inversion()` tool to perform ensemble Kalman inversion:
+```julia
+initial_ensemble = construct_initial_ensemble(prior, N_ensemble)
+ensemble_kalman_inversion =
+    EnsembleKalmanProcess(initial_ensemble, y, Γ, Inversion())
+```
+Then we iterate...
+```julia
+N_iterations = 5
+for i in 1:N_iterations`
+    # get the latest parameter ensemble
+    params_i = get_phi_final(prior, ensemble_kalman_process)
+    # run a simulation for each parameter in the ensemble
+    G_ens = hcat([G(params_i[:, i]) for i in 1:N_ensemble]...)
+    # perform the Kalman update, producing a new ensemble
+    update_ensemble!(ensemble_kalman_process, G_ens)
+end
+```
+The solution of the ensemble Kalman inversion is the final ensemble of parameters. The results are depicted in \autoref{fig:sinusoid}, by evaluating f at these parameters. We see how these methods will calibrate model parameters without estimating the state, as the random shift is not learnt.
+
+![Sinusoids produced from initial and final ensembles, and the (un-noised) truth. \label{fig:sinusoid}](sinusoid_output.pdf){width=80%}
```

Comment:
```suggestion
![Sinusoids produced from initial and final ensembles, and the (denoised) truth. \label{fig:sinusoid}](sinusoid_output.pdf){width=80%}
```

Reply:
"Denoised" suggests that noise was removed. Maybe "non-noisy"?

Reply:
We are denoising it, if we consider the data to be the truth. I guess truth is a tricky word. Do we mean true dynamics = truth, or observation = truth? Typically in ML, people use the latter